### DISCUSSION TITLE
Sliding Window Inference with different I/O shape

### DISCUSSION THREAD

**Discussion by FrancescoLR** [Project-MONAI/MONAI#1595] [Q&A] [answered]
Hi,

I am using a U-Net like architecture with no padding, which therefore gives in output a patch with a reduced size compared to the input. Now at inference time I would like to use the Sliding Window Inference, but I see that it requires the prediction to have the same shape as the input.. Has anyone experienced this issue? What do you suggest doing?
If I recall correctly this was implemented in Niftynet.

Thanks a lot.

**Comment by Nic-Ma** [comment, marked as answer]
Hi @wyli and @ericspod ,

Could you please share some comments and confirm the feature of NiftyNet?

Thanks in advance.

**Reply by ericspod** [reply]
The issue with this inferer is that it breaks the input up into windows which the network predicts a result for, the expectation is that the result represents a prediction in the same position in space as the input. If the output's dimensions don't match the input this doesn't really hold true anymore unless you further assume that the center of the output is in the same spatial position as that of the input. In this case the output would have to be padded to be able to be pasted back into the reconstructed image/volume, so in your case I think to use this inferer you will have to pad your output. One way to do this is define a new subclass of `nn.Module` which has as an attribute your actual UNet object and has a forward method which uses that object for inferenence then pads the result to match the input.

**Reply by FrancescoLR** [reply]
Alright, I see. Thanks for your kind and detailed explanation Eric!
